Tests/test_BWA_tool.py

```diff
@@ -97,9 +97,9 @@ def test_index(self):
             self.assertTrue(os.path.exists(index_file),
                             "Index File %s not found"
                             % (index_file))
-        self.assertTrue(stdout.startswith("[bwt_gen]"),
-                        "FASTA indexing failed:\n%s\nStdout:%s"
-                        % (cmdline, stdout))
+        self.assertTrue('Finished constructing BWT' in str(stdout) + str(stderr),
```

Comment:
I'm pretty sure ``stdout`` and ``stderr`` are both strings, so you don't need the ``str(...)`` here.

Reply:
This is required because either can be none.

```
$ python test_BWA_tool.py 
test_index (__main__.BwaTestCase)
Test for creating index files for the reference genome fasta file ... ok
test_sampe (__main__.BwaTestCase)
Test for generating samfile by paired end sequencing. ... ERROR
test_samse (__main__.BwaTestCase)
Test for single end sequencing. ... ERROR

======================================================================
ERROR: test_sampe (__main__.BwaTestCase)
Test for generating samfile by paired end sequencing.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_BWA_tool.py", line 150, in test_sampe
    self.do_aln(self.infile1, self.saifile1)
  File "test_BWA_tool.py", line 111, in do_aln
    self.assertTrue("fail to locate the index" not in stderr + stdout,
TypeError: cannot concatenate 'str' and 'NoneType' objects

======================================================================
ERROR: test_samse (__main__.BwaTestCase)
Test for single end sequencing.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_BWA_tool.py", line 132, in test_samse
    self.do_aln(self.infile1, self.saifile1)
  File "test_BWA_tool.py", line 111, in do_aln
    self.assertTrue("fail to locate the index" not in stderr + stdout,
TypeError: cannot concatenate 'str' and 'NoneType' objects

----------------------------------------------------------------------

```

Reply:
Ah. That stems from where we use ``subprocess`` and the ``.communicate(...)`` method, which says it will return ``None`` if capturing that stream was not requested. This could be clearer in our ``Bio.Application`` documentation.

However, here we do request both streams (the default in the ``Bio.Application`` wrapper's ``__call__`` method), so maybe the subprocess docs are also incomplete? I am puzzled, but understand why you used ``str(...)`` now.